### PR TITLE
Add user-agent field to api logging

### PR DESCRIPTION
**Pull request by addjuarez**
Signed-off-by: addjuarez <addiajuarez@gmail.com>

# Description

Capture user-agent header from api calls in api logs.
`gRPC API Called: /dapr.proto.runtime.v1.Dapr/InvokeActor Source: grpc-dotnet/2.47.0 (.NET Core; CLR 3.1.417; netstandard2.1; linux; x64)`

## Issue reference

Please reference the issue this PR will close: #5105 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/grpc/server.go [hunk 1]
```diff
@@ -318,8 +319,14 @@ func shouldRenewCert(certExpiryDate time.Time, certDuration time.Duration) bool
...
 	return func(ctx context.Context, req interface{}, info *grpcGo.UnaryServerInfo, handler grpcGo.UnaryHandler) (interface{}, error) {
+		userAgent := "unknown"
+		if meta, ok := metadata.FromIncomingContext(ctx); ok {
+			if val, ok := meta["user-agent"]; ok {
+				userAgent = val[0]
+			}
+		}
 		if s.infoLogger != nil && info != nil {
-			s.infoLogger.Info("gRPC API Called: ", info.FullMethod)
+			s.infoLogger.Infof("gRPC API Called: %s Source: %s", info.FullMethod, userAgent)
```
Comment (halspang) [inline pkg/grpc/server.go:329]: I think it'd be better if we just used the name `UserAgent` in the log. It's a known term.

File: pkg/http/server.go [hunk 2]
```diff
@@ -228,7 +228,12 @@ func (s *server) useMetrics(next fasthttp.RequestHandler) fasthttp.RequestHandle
 
 func (s *server) apiLoggingInfo(next fasthttp.RequestHandler) fasthttp.RequestHandler {
 	return func(ctx *fasthttp.RequestCtx) {
-		infoLog.Infof("HTTP API Called: %s %s", ctx.Method(), ctx.Path())
+		userAgent := string(ctx.Request.Header.Peek("User-Agent"))
+		if userAgent == "" {
+			userAgent = "unknown"
+		}
+
+		infoLog.Infof("HTTP API Called: %s %s Source: %s", ctx.Method(), ctx.Path(), userAgent)
```
Comment (halspang) [inline pkg/http/server.go:236]: Same here. `Source` -> `UserAgent`

**Review by halspang — CHANGES_REQUESTED** [review]

**Review by halspang — APPROVED** [review]

**Review by artursouza — COMMENTED** [review]
E2E tests?

**Comment by halspang** [comment]
>  E2E tests?

@artursouza - Do we have the ability to e2e test logs? I'm not seeing anything that already exists.

**Comment by artursouza** [comment]
> > E2E tests?
> 
> @artursouza - Do we have the ability to e2e test logs? I'm not seeing anything that already exists.

I think quickstarts now.

**Pull request merged**